src/main/java/htsjdk/samtools/seekablestream/SeekableStream.java

```diff
@@ -68,4 +86,38 @@ public void readFully(byte b[]) throws IOException {
         }
     }
 
+    /**
+     * Mark the current position of the stream.
+     *
+     * <p>Note: there is no limit for reading.
+     *
+     * @param readlimit ignored.
+     */
+    @Override
+    public final synchronized void mark(int readlimit) {
+        try {
+            mark = OptionalLong.of(position());
```

Comment:
in reviewing this we were concerned about the scenario where position legitimately throws an exception causing mark to remain un-updated. In this cas, due to the catch, it will lead to a silent error.

There are several ways around this:
1. throw an unchecked exception instead
2. look at the specific exceptions thrown by current implementations and work with these
3. modify contract for position() to not throw, and simultaneously modify known implementations.

if you have other solutions, please discuss.

Reply:
I opted for a combination of 1 and 2: if the exception is a `ClosedChannelException`, there's no effect at all (as the contract suggests); otherwise, it throws an `RuntimeIOException` for converting into an unchecked exception.